src/marqo/core/index_management/index_management.py

```diff
@@ -544,16 +552,32 @@ def _delete_index_settings_by_name(self, index_name: str):
     def _deployment_lock_context_manager(self):
         """A context manager for deployment lock acquisition.
 
+        If the _enable_index_operations flag is set to True, the context manager tries to acquire the deployment lock.
+            If the lock is acquired, the context manager yields control to the caller. Or if the lock is not acquired,
+            it raises an OperationConflictError.
+            If the lock is None, the context manager yields control to the caller without acquiring the lock.
+        If the _enable_index_operations flag is set to False, the context manager raises an InternalError during
+            index operations.
+
         Raises:
             OperationConflictError: If another index creation/deletion operation is
                 in progress and the lock cannot be acquired
-            InternalError: If deployment lock is not instantiated
-        """
-        if self._zookeeper_deployment_lock is None:
-            raise InternalError("Deployment lock is not instantiated and cannot be used for index creation/deletion")
-        try:
-            with self._zookeeper_deployment_lock:
-                yield
-        except ZookeeperLockNotAcquiredError:
-            raise OperationConflictError("Another index creation/deletion operation is in progress. "
-                                         "Your request is rejected. Please try again later")
\ No newline at end of file
+            InternalError: If index_management object is not enabled for index operations
+        """
+        if self._enable_index_operations:
+            if self._zookeeper_deployment_lock is None:
+                logger.warning(f"You are trying to perform an index operation without setting a Zookeeper client. "
```

Comment:
I suggest a much shorter warning. Note this is mainly for us not the user. Something like "No Zookeeper client provided. Concurrent operations could lead to a race condition". You can include a docs link to be very thorough

Reply:
done